.github/workflows/release.yaml

```diff
@@ -0,0 +1,33 @@
+# Default release on push in main
+name: Release
+on:
+  push:
+    branches:
+    - main
+    
+jobs:
+ release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      
+    - name: Retrieve version
+      id: version 
+      run: | 
+        git fetch --tags
+        RELEASE_TAG=$(git tag | sort -g | tail -1 | awk -F. -v OFS=. '{$NF=$NF+1;print}')
```

Comment:
`sort`will not handle the ordering 'v0.9.0' and 'v0.10.0' for example, we probably need something more complex if the patch or minor version is more than 9.
It looks that `sort -V` is doing the job:
```
sort -V << EOF
v0.0.11
v0.0.1
v0.0.2
v0.0.9
v0.0.10
v0.0.19
v0.0.20
EOF

shows
v0.0.1
v0.0.2
v0.0.9
v0.0.10
v0.0.11
v0.0.19
v0.0.20
```

And a minor remark, here you increment the patch version but the acceptance criteria says to increment the minor.

Reply:
I fixed the issues. The awk for minor looks cleaner now by the way:
`$(git tag | sort -V | tail -1 | awk -F. '{print $1"."($2+1)"."$3}'`
For major we can use:
`$(git tag | sort -V | tail -1 | awk -F. '{print ($1+1)"."$2"."$3}'`
and patch:
`$(git tag | sort -V | tail -1 | awk -F. '{print $1"."$2"."($3+1)}'`

I agree about the improvements based on the commit message we update the right number (or no number), but for a first release I would already go with that.